### PR TITLE
Add difference method

### PR DESCRIPTION
**Pull request by cquiroz**
For the seqexec I added a method to calculate the minimum angle difference between 2 angles
This PR adds this code to gsp-math and eventually we'd call it from the seqexec

### CODE REVIEW

File: modules/math/shared/src/main/scala/gsp/math/Angle.scala [hunk 1]
```diff
@@ -110,6 +110,11 @@ sealed class Angle protected (val toMicroarcseconds: Long) {
   override final def hashCode =
     toMicroarcseconds.toInt
 
+  /**
+   * angle difference or separation between two angles in the range [0 .. π)
```
Comment (tpolecat) [inline modules/math/shared/src/main/scala/gsp/math/Angle.scala]: [0 .. π]

File: modules/math/shared/src/main/scala/gsp/math/Angle.scala [hunk 2]
```diff
@@ -236,6 +241,17 @@ object Angle extends AngleOptics {
       degrees.toLong         * 1000 * 1000 * 60 * 60
     )
 
+  /**
+   * Calculate the angle difference or separation between two angles.
+   * The calculation is such that you get the minimal angle in the range [0 .. π)
```
Comment (tpolecat) [inline modules/math/shared/src/main/scala/gsp/math/Angle.scala]: [0 .. π]

File: modules/tests/shared/src/test/scala/gsp/math/AngleSpec.scala [hunk 3]
```diff
@@ -135,4 +134,29 @@ final class AngleSpec extends CatsSuite {
...
+    }
+  }
+
+  test("Difference with reverse is 180") {
+    forAll { (a: Angle) =>
+      a.difference(a + Angle.Angle180) shouldEqual Angle.Angle180
+    }
+  }
+
+  test("Difference is in the range [0 .. π)") {
```
Comment (tpolecat) [inline modules/tests/shared/src/test/scala/gsp/math/AngleSpec.scala]: [0 .. π] … this test should fail for (0, 180)

File: modules/tests/shared/src/test/scala/gsp/math/AngleSpec.scala [hunk 4]
```diff
@@ -135,4 +134,29 @@ final class AngleSpec extends CatsSuite {
...
+    implicit val order: Order[Angle] = Angle.AngleOrder
+    forAll { (a: Angle, b: Angle) =>
+      (a.difference(b) >= Angle.Angle0 && a.difference(b) < Angle.Angle180) shouldBe true
+    }
+  }
+
+  test("Difference is commutative") {
+    forAll { (a: Angle, b: Angle) =>
+      a.difference(b) shouldEqual a.difference(b)
```
Comment (tpolecat) [inline modules/tests/shared/src/test/scala/gsp/math/AngleSpec.scala]: One of these should be `b.difference(a)` ;-)

File: modules/tests/shared/src/test/scala/gsp/math/AngleSpec.scala (hunk 3 above)
Comment (cquiroz) [reply modules/tests/shared/src/test/scala/gsp/math/AngleSpec.scala]: Are you using `[]` for an inclusive range, i.e `0 <= x <= pi`?

File: modules/tests/shared/src/test/scala/gsp/math/AngleSpec.scala (hunk 4 above)
Comment (cquiroz) [reply modules/tests/shared/src/test/scala/gsp/math/AngleSpec.scala]: Good catch

File: modules/tests/shared/src/test/scala/gsp/math/AngleSpec.scala (hunk 3 above)
Comment (tpolecat) [reply modules/tests/shared/src/test/scala/gsp/math/AngleSpec.scala]: Yes, `[` for inclusive, `(` for exclusive.

**Comment by tpolecat** [comment]
👍

**Pull request merged**